docs/src/simulation_tips.md

```diff
@@ -0,0 +1,153 @@
+significant.
+
+
+
+## GPU simulation tips
+
+Running on GPUs is very different from running on CPUs. Oceananigans makes most of the necessary
+changes in the background, so that for very simple simulations changing between CPUs and GPUs is
+just a matter of changing the `architecture` argument in the model from `CPU()` to `GPU()`. However,
+for more complex simulations some care needs to be taken on the part of the user. While knowledge of
+GPU computing (and Julia) is again desirable, the inexperienced user can also achieve efficient GPU
+simulations by following a few simple principles.
+
+
+### Variables that need to be used in GPU computations need to be defined as constants
+
+Any global variable that needs to be accessed by the GPU needs to be a constant or the simulation will crash
+
+
+### Complex diagnostics using `ComputedField`s may not work on GPUs
+
+`ComputedField`s are the most convenient way to calculate diagnostics for your simulation. They will
+always work on CPUs, but when their complexity is high (in terms of number of abstract operations)
+the compiler can't  translate them into GPU code and they fail. For example, in the example below,
+calculating `u²` works in both CPUs and GPUs, but calculating `KE` only works in CPUs:
+
+```julia
+u, v, w = model.velocities
+u² = ComputedField(u^2)
+KE = ComputedField((u^2 + v^2 + w^2)/2)
+compute!(u²)
+compute!(KE)
+```
+
+There are two approaches to bypass this issue. The first is to nest `ComputedField`s. For example,
+we can make `KE` be computed on GPU by defining it as
+```julia
+u, v, w = model.velocities
+u² = ComputedField(u^2)
+v² = ComputedField(v^2)
+w² = ComputedField(w^2)
+uplusv = ComputedField(u² + v²)
+KE = ComputedField((uplusv + w²)/2)
+compute!(KE)
+```
+
+This is a simple workaround that is especially suited for the development stage of a simulation.
+However, when running this on a GPU, the code will iterate over the whole domain 5 times to
+calculate `KE` (one for each computed field defined), which is not very efficient.
+
+A different way to calculate `KE` is by using `KernelComputedField`s, where the
+user manually specifies the computing kernel to the compiler. The advantage of this method is that
+it's more efficient (the code will only iterate once over the domain in order to calculate `KE`),
+but the disadvantage is that this requires that the has some knowledge of Oceananigans operations
+and how they should be performed on a C-grid. For example calculating `KE` with this approach would
+look like this:
+
+```julia
+using Oceananigans.Operators
+using KernelAbstractions: @index, @kernel
+using Oceananigans.Grids: Center, Face
+using Oceananigans.Fields: KernelComputedField
+
+@inline ψ²(i, j, k, grid, ψ) = @inbounds ψ[i, j, k]^2
+
+@kernel function kinetic_energy_ccc!(tke, grid, u, v, w)
+    i, j, k = @index(Global, NTuple)
+    @inbounds tke[i, j, k] = (
+                              ℑxᶜᵃᵃ(i, j, k, grid, ψ², u) + # Interpolate x to center
```

Comment:
Should the comments be changed to describe the functio to be interpolated?

Reply:
Done!